boost-maven/boost-maven-plugin/src/main/java/io/openliberty/boost/liberty/LibertyPackageMojo.java

```diff
@@ -78,51 +84,71 @@ public void execute() throws MojoExecutionException {
-                    addSpringBootVersionToManifest(springBootVersion);
-                }
-            } else { // Dealing with an EE based app
+            // Check if we need to attach based on the classifier configuration
+            if (MavenProjectUtil.isNotNullOrEmpty(springBootClassifier)) {
                 attach = false;
-                boosterFeatures = getBoosterConfigsFromDependencies(project);
-                generateServerXMLJ2EE(boosterFeatures);
-                installMissingFeatures();
-                installApp("project");
+            } else {
+                attach = true;
+            }
 
+            File springBootUberJar = net.wasdev.wlp.maven.plugins.utils.SpringBootUtil.getSpringBootUberJAR(project,
+                    getLog());
+            validateSpringBootUberJAR(springBootUberJar);
+            copySpringBootUberJar(springBootUberJar, attach); // Only copy back if we need to overwrite the project
+                                                              // artifact
+
+            generateServerConfigSpringBoot();
+            
+            installMissingFeatures();
+            installApp("spring-boot-project");
+
+            if (springBootUberJar != null) {
+                // Create the Liberty Uber JAR from the Spring Boot Uber JAR in place
+                createUberJar(springBootUberJar.getAbsolutePath(), attach);
+            } else {
+                // The Spring Boot Uber JAR was already replaced with the Liberty Uber JAR (this
+                // is a re-execution in the non-classifier scenario)
                 createUberJar(null, attach);
             }
-        } catch (TransformerException | ParserConfigurationException e) {
+
+            if (!MavenProjectUtil.isNotNullOrEmpty(springBootClassifier)) {
+                // If necessary, add the manifest to prevent Spring Boot from repackaging again
+                addSpringBootVersionToManifest(springBootVersion);
+            }
+        } else { // Dealing with an EE based app
+            attach = false;
+            boosterFeatures = getBoosterConfigsFromDependencies(project);
+            generateServerXMLJ2EE(boosterFeatures);
+            installMissingFeatures();
+            installApp("project");
+
+            createUberJar(null, attach);
+        }
+    }
+    
+    /**
+     * Generate config for the Liberty server based on the Maven Spring Boot
+     * project. 
+     * 
+     * @throws MojoExecutionException
+     */
+    private void generateServerConfigSpringBoot() throws MojoExecutionException {
+        
+        try {
+            // Get Spring Boot starters from Maven project
+            List<String> springBootStarters = MavenProjectUtil.getSpringBootStarters(project);
+            
+            // Set Liberty server path
+            String libertyServerPath = projectBuildDir + "/liberty/wlp/usr/servers/" + libertyServerName; 
```

Comment:
Can we set the install directory (`projectBuildDir + "/liberty/wlp"`) as a constant since we reference it in several places? It might change or be configurable in the future.